src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -583,6 +583,12 @@ TER DirectStepI::check (StrandContext const& ctx) const
         return temBAD_PATH;
     }
 
+    if (src_ == dst_)
+    {
+        JLOG (j_.debug()) << "DirectStepI: same src and dst.";
+        return temBAD_PATH;
```

Comment:
Never mind.  We're returning a `tem` immediately above here.  If that one is okay then this one is okay.